Add language to MVP.md about failure modes (#134)
WebAssembly/gc: proposals/gc/MVP.md

* The so-defined runtime type is the only type information that can be discovered about a reference value at runtime; a structure or array with RTT `any` thereby is fully opaque to runtime type checks (and an implementation may choose to optimize away its RTT).

#### Failure

Allocation of memory objects may fail for implementation-specific reasons, such as system resource exhaustion or a host-defined limit, or being unable to fit a large array in available address space.

> Handling of an allocation failure may be handled in an implementation-defined way, such as trapping or terminating the host process. Implementors should be aware this may be used as a denial of service vector.

Reviewer: ```suggestion
Similar to stack overflow, allocation failure may materialise in an implementation-dependent way, such as trapping or terminating the host process. Implementors should be aware this may be used as a denial of service vector.
```

* `struct.new_with_rtt <typeidx>` allocates a structure with RTT information determining its [runtime type](#values) and initialises its fields with given values
  - `struct.new_with_rtt $t : [t'* (rtt n $t)] -> [(ref $t)]`
    - iff `$t = struct (mut t')*`

>   - allocation failure behavior is [implementation-defined](#failure)

Reviewer: ```suggestion
  - allocation failure behavior is [implementation-dependent](#failure)
```

* `struct.new_default_with_rtt <typeidx>` allocates a structure of type `$t` and initialises its fields with default values
  - `struct.new_default_with_rtt $t : [(rtt n $t)] -> [(ref $t)]`
    - iff `$t = struct (mut t')*`
    - and all `t'*` are defaultable

>   - allocation failure behavior is [implementation-defined](#failure)

Reviewer: ```suggestion
  - allocation failure behavior is [implementation-dependent](#failure)
```

* `array.new_with_rtt <typeidx>` allocates an array with RTT information determining its [runtime type](#values)
  - `array.new_with_rtt $t : [t' i32 (rtt n $t)] -> [(ref $t)]`
    - iff `$t = array (var t')`

>   - allocation failure behavior is [implementation-defined](#failure)

Reviewer: ```suggestion
  - allocation failure behavior is [implementation-dependent](#failure)
```

* `array.new_default_with_rtt <typeidx>` allocates an array and initialises its fields with the default value
  - `array.new_default_with_rtt $t : [i32 (rtt n $t)] -> [(ref $t)]`
    - iff `$t = array (var t')`
    - and `t'` is defaultable

>   - allocation failure behavior is [implementation-defined](#failure)

Reviewer: ```suggestion
  - allocation failure behavior is [implementation-dependent](#failure)
```
